astropy/table/serialize.py

```diff
@@ -130,17 +136,32 @@ def _represent_mixin_as_column(col, name, new_cols, mixin_cols,
         else:
             new_name = name + '.' + data_attr
 
+        # If the data attribute is not a Mixin (aka a Column or ndarray) and
+        # multidimensional. For instance saving an N-d Time object can send a
+        # data attribute that is an N-d np.ndarray, so we need to recast that
+        # as NdarrayMixin so it gets flattened (in the right context i.e. ECSV).
+        if (not has_info_class(data, MixinInfo)
+                and col.info._serialize_context in FLATTEN_MULTIDIM
+                and len(data.shape) > 1):
+            data = NdarrayMixin(data)
+
         if not has_info_class(data, MixinInfo):
             col_cls = MaskedColumn if (hasattr(data, 'mask')
                                        and np.any(data.mask)) else Column
+            # TODO: I think the **info here is not really needed and tests pass
+            # without it. This just puts extra stuff into the header as far as I
+            # can tell. Remove?? Will this be an API change?
             new_cols.append(col_cls(data, name=new_name, **info))
             obj_attrs[data_attr] = SerializedColumn({'name': new_name})
         else:
             # recurse. This will define obj_attrs[new_name].
             _represent_mixin_as_column(data, new_name, new_cols, obj_attrs)
             obj_attrs[data_attr] = SerializedColumn(obj_attrs.pop(new_name))
 
-        # Strip out from info any attributes defined by the parent
+        # TODO: the next two blocks don't seem to belong in this loop, so move
```

Comment:
Will do this TODO in the next set of commits. Hopefully if it passes tests then good to go.